api/dao/projectDAO.js

```diff
@@ -0,0 +1,645 @@
+    newProject.read = ['sysadmin', 'staff'];
+    newProject.write = ['sysadmin', 'staff'];
+    newProject.delete = ['sysadmin', 'staff'];
+    projectData._createdBy = user;
+    projectData.createdDate = Date.now();
+
+    if (projectLegislationYear == 2018) 
+    {
+      newProject.legislation_2018 = projectData;
+    } 
+    else if (projectLegislationYear == 2002) 
+    {
+      newProject.legislation_2002 = projectData;
+    } 
+    else if (projectLegislationYear == 1996) 
+    {
+      newProject.legislation_1996 = projectData;
+    }
+
+    // Currently this will save based on the entire project model.
+    // Meaning there will be three project legislation keys ( legislation_1996, legislation_2002, legislation_2018) only one of which will be populated with data.
+    // The other two keys will be full of null values, as well as any other fields that are in the project model and are not explicitly defined above.
+    return newProject.save()
+                    .then(function (createdProject) 
+                    {
+                        Utils.recordAction('Post', 'Project', user, createdProject._id);
+
+                        return createdProject;
+                    })
+                    .catch(function (err) {
+                        throw Error('Failed to create project: ', err);
+                    });
+};
+
+exports.updateProject = async function(user, sourceProject, updatedProject)
+{
+    let projectLegislationYear;
+    let filteredData;
+
+    // if project legislation doesn't exist then look up current legislation for the project
+    if (updatedProject.legislationYear) 
+    {
+      projectLegislationYear = updatedProject.legislationYear;
+      // check if the passed in project year exists in the legislation year list
+      if (!sourceProject.legislationYearList.includes(projectLegislationYear))
+      {
+        sourceProject.legislationYearList.push(projectLegislationYear);
+      }
+    } 
+    else 
+    {
+      // look up the current project legislation
+      projectLegislationYear = sourceProject.currentLegislationYear.split("_")[1];
+    }
+
+    if (projectLegislationYear == 2018) 
+    {
+      filteredData = sourceProject.legislation_2018;
+      filteredData.legislation = "2018 Environmental Assessment Act";
+    } 
+    else if (projectLegislationYear == 2002) 
+    {
+      filteredData = sourceProject.legislation_2002;
+      filteredData.legislation = "2002 Environmental Assessment Act";
+    } 
+    else if (projectLegislationYear == 1996) 
+    {
+      filteredData = sourceProject.legislation_1996;
+      filteredData.legislation = "1996 Environmental Assessment Act";
+    }
+
+    if (!filteredData) 
+    {
+      defaultLog.info("Couldn't find that object!");
+      return Actions.sendResponse(res, 404, {});
+    }
+
+    delete updatedProject.read;
+    delete updatedProject.write;
+    delete updatedProject.delete;
+
+    filteredData.type              = updatedProject.type;
```

Comment:
This can be spread in and then only the properties that are different need to be manually set. 
`filteredData = { 
    ...updatedProject,
    filteredData.decisionDate = updatedProject.decisionDate ? new Date(updatedProject.decisionDate) : null,
    etc etc,
}`